src/transformers/trainer.py

```diff
@@ -1215,6 +1215,10 @@ def train(
                     break
 
         for epoch in range(epochs_trained, num_train_epochs):
+            if self.use_amp and hasattr(self, "qat_active") and callable(self.qat_active) and self.qat_active(epoch):
+                logger.info("entering QAT phase, disabling FP16 training")
```

Comment:
The first time when we enter this QAT mode from pruning, the `self.optimizer` is still an AdamW, and I think we need to create it as ScheduledOptimizer (as done in `create_optimizer` of this class).

Reply:
instead of setting `ues_amp` to `False`, I just disabled the scaler here so that way the optimizer is still called through the wrapped scaler and the base trainer doesn't need to call back into SparseML